llvm/lib/Target/NVPTX/NVPTXUtilities.h

```diff
@@ -85,8 +85,14 @@ inline unsigned promoteScalarArgumentSize(unsigned size) {
 
 bool shouldEmitPTXNoReturn(const Value *V, const TargetMachine &TM);
 
-inline bool Isv2x16VT(EVT VT) {
-  return (VT == MVT::v2f16 || VT == MVT::v2bf16 || VT == MVT::v2i16);
+inline bool isPackedVectorTy(EVT VT) {
+  return (VT == MVT::v4i8 || VT == MVT::v2f16 || VT == MVT::v2bf16 ||
+          VT == MVT::v2i16 || VT == MVT::v2f32);
+}
+
+inline bool isPackedElementTy(EVT VT) {
```

Comment:
What makes these types 'packed' ? The function could use a comment explaining what it means.
It clearly not the common meaning of packed as related to aligning struct fields w/o alignment gaps.

Does it mean that it's one of the types for which we have vectorized PTX instructions? It could use a more distinct name.

Reply:
"Packed" here refers to whether a single vector type or two or more element types can fit into a single register.